atsc/src/compressor/rle.rs

```diff
@@ -0,0 +1,335 @@
+/*
+Copyright 2024 NetApp, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use crate::{
+    compressor::CompressorResult,
+    optimizer::utils::{Bitdepth, DataStats},
+};
+
+use super::BinConfig;
+use bincode::{Decode, Encode};
+use log::{debug, trace};
+use std::collections::BTreeMap;
+
+const RLE_COMPRESSOR_ID: u8 = 60;
```

Comment:
What is this for?
If you want to use this in the future adding new subkinds of RLE compression, lets use an enum instead.

Reply:
right, should still be an enum, but I wont block on it, since this is how other compressors are.